open-hackathon-adminUI/src/app/admin/login.py

```diff
@@ -0,0 +1,161 @@
+__author__ = 'root'
+
+import sys
+import urllib2
+sys.path.append("..")
+# -*- coding:utf8 -*-
+# encoding = utf-8
+from app.functions import get_remote, get_config, post_to_remote
+from app.log import log
+import json
+from user_mgr import user_manager
+from flask import redirect, url_for
+
+hackathon_api_url = get_config("hackathon-api/endpoint")
+hackathon_name = get_config("javascript/hackathon/name")
+
+
+
+
+class LoginBase():
+
+    def login2db(self,args):
+
+        access_token = args.get('access_token')
+        provider = args.get('provider')
+        # get user info
+
+        user_info_resp = get_remote(get_config('login/github/user_info_url') + access_token)
```

Comment:
refresh the whole logic implementation about login operation
gitcafe , QQ adn github are took into account